apps/chat/src/store/application/application.epics.ts

```diff
@@ -0,0 +1,202 @@
+  action$.pipe(
+    filter(ApplicationActions.create.match),
+    switchMap(({ payload: { applicationData, applicationName } }) => {
+      if (!applicationData.version || !applicationData.iconUrl) {
+        // TODO: update types or add fail epic
+        return EMPTY;
+      }
+
+      const apiFormattedData: CreateApplicationModel = {
+        endpoint: applicationData.completionUrl,
+        display_name: applicationData.name,
+        display_version: applicationData.version,
+        icon_url: applicationData.iconUrl,
+        description: applicationData.description,
+        features: applicationData.features,
+        input_attachment_types: applicationData.inputAttachmentTypes,
+        max_input_attachments: applicationData.maxInputAttachments,
+        defaults: {},
+      };
+
+      return ApplicationService.create(applicationName, apiFormattedData).pipe(
+        switchMap((application: ApplicationListItemModel) =>
+          ApplicationService.get(application.url).pipe(
+            map((application) => {
+              return ModelsActions.addModels({
+                models: [application],
+              });
+            }),
+          ),
+        ),
+        catchError((err) => {
+          console.error(err);
+          return of(ApplicationActions.createFail());
+        }),
+      );
+    }),
+  );
+
+const createFailEpic: AppEpic = (action$) =>
+  action$.pipe(
+    filter(ApplicationActions.createFail.match),
+    switchMap(() =>
+      of(UIActions.showErrorToast(translate(errorsMessages.createFailed))),
+    ),
+  );
+
+const deleteApplicationEpic: AppEpic = (action$) =>
+  action$.pipe(
+    filter(ApplicationActions.delete.match),
+    switchMap((action: DeleteApplicationAction) =>
+      ApplicationService.delete(action.payload.currentEntityName).pipe(
+        switchMap(() => {
+          return of(
+            ApplicationActions.deleteSuccess(),
+            ModelsActions.deleteModel({
+              modelName: action.payload.currentEntityName,
+              id: action.payload.currentEntityId,
+            }),
+          );
+        }),
+        catchError((err) => {
+          console.error(err);
+          // Dispatch fail action
+          return of(ApplicationActions.deleteFail());
+        }),
+      ),
+    ),
+  );
+
+// Fetch listings epic
+const listApplicationsEpic: AppEpic = (action$) =>
+  action$.pipe(
+    filter(ApplicationActions.list.match),
+    switchMap(() =>
+      ApplicationService.listing().pipe(
+        switchMap((applications) =>
+          of(ApplicationActions.listSuccess(applications)),
+        ),
+        catchError((err) => {
+          console.error(err);
+          return of(ApplicationActions.listFail());
+        }),
+      ),
+    ),
+  );
+
+const moveApplicationEpic: AppEpic = (action$) =>
+  action$.pipe(
+    filter(ApplicationActions.move.match),
+    switchMap(({ payload }) => {
+      if (payload.oldApplicationName !== payload.applicationData.name) {
```

Comment:
The application may lie in some folder if someone creates it directly through api. We have to be careful and handle this.
Let's try to keep `folderId` and change only `name` (the last part from id).

Now the service creates the path with `apiKey`, `bucket` and `name`, so this may lead to the structure overriding